### PR TITLE
nixUnstable: pre7346_5e7ccdc9 -> pre7534_b92f58f6

### PR DESCRIPTION
**Pull request by nlewo**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/package-management/nix/default.nix [hunk 1]
```diff
@@ -39,7 +41,8 @@ common =
 
       nativeBuildInputs =
         [ pkgconfig ]
-        ++ lib.optionals is24 [ autoreconfHook autoconf-archive bison flex libxml2 libxslt docbook5 docbook_xsl_ns jq ];
+        ++ lib.optionals is24 [ autoreconfHook autoconf-archive bison flex libxml2 libxslt
+                                docbook5 docbook_xsl_ns jq gmock git mercurial ];
```
Comment (edolstra) [inline pkgs/tools/package-management/nix/default.nix:45]: Don't add git and mercurial, we already have way too many dependencies.

File: pkgs/tools/package-management/nix/default.nix [hunk 2]
```diff
@@ -188,18 +191,18 @@ in rec {
...
-      rev = "5e7ccdc9e3ddd61dc85e20c898001345bfb497a5";
-      sha256 = "10jg0rq92xbigbbri7harn4b75blqaf6rjgq4hhvlnggf2w9iprg";
+      rev = "b92f58f6d9e44f97002d1722bd77bad939824c1c";
+      sha256 = "1p791961y5v04kpz37g6hm98f1ig7i34inxl9dcj3pbqhf5kicxg";
     };
 
     crates = fetchurl {
-      url = "https://hydra.nixos.org/build/115942497/download/1/nix-vendored-crates-2.4pre20200403_3473b19.tar.xz";
-      sha256 = "a83785553bb4bc5b28220562153e201ec555a00171466ac08b716f0c97aee45a";
+      url = "https://hydra.nixos.org/build/118797694/download/1/nix-vendored-crates-2.4pre7534_b92f58f6.tar.xz";
```
Comment (edolstra) [inline pkgs/tools/package-management/nix/default.nix:204]: Note that this file could be garbage-collected in the future if you don't mark it as "Keep" in the Hydra web interface.

Comment (nlewo) [reply pkgs/tools/package-management/nix/default.nix:204]: I don't have an account on this Hydra so I don't think I can do it.

File: pkgs/tools/package-management/nix/default.nix (hunk 1 above)
Comment (nlewo) [reply pkgs/tools/package-management/nix/default.nix]: Done.

File: pkgs/tools/package-management/nix/default.nix (hunk 2 above)
Comment (nlewo) [reply pkgs/tools/package-management/nix/default.nix:204]: Here is the build link https://hydra.nixos.org/build/118797694#tabs-summary ;)

Comment (edolstra) [reply pkgs/tools/package-management/nix/default.nix:204]: Done.

**Comment by nlewo** [comment]
Thanks!

**Pull request merged**